app/assets/javascripts/discourse/widgets/header-topic-info.js

```diff
@@ -112,11 +116,7 @@ export default createWidget("header-topic-info", {
     const category = topic.get("category");
 
     if (loaded || category) {
-      if (
-        category &&
-        (!category.get("isUncategorizedCategory") ||
-          !this.siteSettings.suppress_uncategorized_badge)
-      ) {
+      if (this.showCategory(topic)) {
```

Comment:
I'm curious why don't we directly use category here ?

Reply:
I like the function name with topic as argument.

> Should we show the category for the given topic?

Reply:
well ok then :+1: